### PR TITLE
Iox #590 Add component diagram for upcoming iceoryx release

### PR DESCRIPTION
**Pull request by mossmaurice**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/advanced/best-practice-for-testing.md

## Notes for Reviewer
* Introduce PlantUML component diagram to outline target architecture for next iceoryx release
* Add scripts which..
  *  .. lists all namespaces used
  * .. analyse binary for blacklisted namespaces

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Closes partly #590

### CODE REVIEW

File: tools/print_all_namespace.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
```
Comment (elfenpiff) [inline tools/print_all_namespace.sh:1]: Traceability. What does this have to do with the component diagram? Please create a separate issue and adjust the commit message iox number to be traceable.

File: iceoryx_examples/icedelivery/CMakeLists.txt [hunk 2]
```diff
@@ -35,7 +35,7 @@ target_compile_options(iox-cpp-publisher PRIVATE ${ICEORYX_WARNINGS} ${ICEORYX_S
```
Comment (elfenpiff) [inline iceoryx_examples/icedelivery/CMakeLists.txt:35]: Traceability. What does this have to do with the component diagram? Please create a separate issue and adjust the commit message iox number to be traceable.

File: iceoryx_examples/icedelivery/CMakeLists.txt [hunk 3]
```diff
@@ -35,7 +35,7 @@ target_compile_options(iox-cpp-publisher PRIVATE ${ICEORYX_WARNINGS} ${ICEORYX_S
 
 add_executable(iox-cpp-publisher-untyped ./iox_publisher_untyped.cpp)
 target_link_libraries(iox-cpp-publisher-untyped
-    iceoryx_posh::iceoryx_posh
+    iceoryx_posh::iceoryx_posh # @todo #590 Should link against iceoryx_runtime (which depends on iceoryx_posh)
```
Comment (elfenpiff) [inline iceoryx_examples/icedelivery/CMakeLists.txt:38]: I thought no todos in code. You could add this todo to the corresponding issue as a todo, then we can keep really track of it.

File: doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+      component helper <<namespace>>
+   }
+   component errors <<namespace>>
+}
+
+' HOOFS: Helpful Objects Optimised For Safety
+' Library containing modern C++ STL constructs
+package iceoryx_hoofs  {
+   frame "hoofs library" {
+      component concurrent <<namespace>>
```
Comment (elfenpiff) [inline doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml:61]: Is this list incomplete on purpose because it is a first draft?

File: doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+      component runtime <<namespace>>
+   }
+
+   frame "gateway library" {
+      component gw <<namespace>>
+   }
+
+   frame "RouDi library" {
+      component roudi <<namespace>>
+      component build <<namespace>>
```
Comment (elfenpiff) [inline doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml:47]: We have a build namespace?

File: doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+' @todo #590
+' RouDi split, separate files and namespaces for static/dynamic discovery
+' Implement exported libraries (add_library in iceoryx_posh/CMakeLists.txt)
+' RouDiEnvironment shall not be in roudi namespace, but in testing
+
+header Eclipse iceoryx component overview
+
+' POSH: POSIX SHared memory
+' Publish/subscribe IPC communication infrastructure
+package iceoryx_posh {
```
Comment (elfenpiff) [inline doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml:29]: What do you think about the following:
- `iceoryx_posh` component handles all shared memory related tasks
- `iceoryx_roudi` is another component which depends of `iceoryx_posh`
- `iceoryx_runtime` is another component as well

File: doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml (hunk 5 above)
Comment (mossmaurice) [reply doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml:47]: Yes, please run the provided script.

File: tools/print_all_namespace.sh (hunk 1 above)
Comment (mossmaurice) [reply tools/print_all_namespace.sh:1]: Nope, script and diagram belong together. This script can be used to update the diagram and make sure not namespaces are missing. We could also automate this at some point.

File: iceoryx_examples/icedelivery/CMakeLists.txt (hunk 2 above)
Comment (mossmaurice) [reply iceoryx_examples/icedelivery/CMakeLists.txt:35]: This is an example motivating an extra `runtime` library and belongs to the diagram.

File: iceoryx_examples/icedelivery/CMakeLists.txt (hunk 3 above)
Comment (mossmaurice) [reply iceoryx_examples/icedelivery/CMakeLists.txt:38]: No todos? :grin: There are 127 ` @ todo` in the iceoryx code base currently. I can move this specific one to the issue.

File: doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml (hunk 6 above)
Comment (mossmaurice) [reply doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml:29]: Yeah, that's a great idea. I'll address this in a follow-up with #539. We need to keep several things in mind. I'll add this to the issue:
|Module| Safety-certified version | Community version |
|---|---|---|
| Service Discovery | Only static with config file | dynamic service discovery |
| Shared Memory | Created statically during init phase | Create objects in shared memory on-the-fly  |

File: tools/print_all_namespace.sh (hunk 1 above)
Comment (dkroenke) [inline tools/print_all_namespace.sh:1]: This script can be combined with https://github.com/eclipse-iceoryx/iceoryx/blob/master/tools/list_stl_dependencies.sh

**Review by ZhenshengLee — COMMENTED** [review]
Good study material.

File: doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+   component publisher.h <<file>>
+   component subscriber.h <<file>>
+   component node.h <<file>>
+   component types.h <<file>>
+   component wait_set.h <<file>>
+   component user_trigger.h <<file>>
+   component cpp2c <<namespace>>
+}
+
+' CALF: Can Also Left (un) Finished
```
[the remark this answers is not in the document]
Comment (elBoberido) [reply doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml:101]: - CALlow Fun
- CAn Leak Fast
- Code And Lots (of) Fun
- Come And Let Forget

File: iceoryx_examples/icedelivery/CMakeLists.txt (hunk 3 above)
Comment (elBoberido) [reply iceoryx_examples/icedelivery/CMakeLists.txt:38]: I think it's fine to have todos in the code as long as they are linked to an issue like this one

File: doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml [hunk 8]
```diff
@@ -0,0 +1,119 @@
+@startuml Eclipse iceoryx Component Overview
```
Comment (elBoberido) [inline doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml:1]: Could you please render this to a SVG? You just need to run `tools/generate_plantuml_svg.sh`

File: doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml (hunk 7 above)
[the remark this answers is not in the document]
Comment (mossmaurice) [reply doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml:101]: Removed `iceoryx_calf` as it will be cumbersome to adapt all the include paths and maintain certain building blocks like e.g. `NamedPipes`, which are only used on Windows in a different package. I added the script `tools/analyse_allowed_namespaces_in_binary.sh` to analyse a binary.

File: doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml (hunk 6 above)
Comment (mossmaurice) [reply doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml:29]: I had a chat yesterday with @elBoberido about it. We concluded that the `iceoryx_roudi` for a default pre-built RouDi makes sense. However, `iceoryx_runtime` will only be a library and not a package in order to be backwards compatible with `v1.0.x`. Having thought a bit more about it today, I realized this will clash with #970 :grimacing: 

@elBoberido What do you think?

File: tools/print_all_namespace.sh (hunk 1 above)
Comment (mossmaurice) [reply tools/print_all_namespace.sh:1]: @dkroenke Let's keep the scripts separate for now and move them into a directory in your follow-up PR.

File: doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml (hunk 8 above)
Comment (mossmaurice) [reply doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml:1]: Done

File: doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml (hunk 6 above)
Comment (mossmaurice) [reply doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml:60]: After discussion with @dkroenke we concluded that sticking to one `iceoryx_posh` CMake package will keep the maintenance overhead lower. I updated the diagram accordingly, introducing new libraries inside `iceoryx_posh`.

File: iceoryx_examples/icedelivery/CMakeLists.txt (hunk 3 above)
Comment (mossmaurice) [reply iceoryx_examples/icedelivery/CMakeLists.txt]: Removed as we want to keep the alias `iceoryx_posh` for backwards compatibility and easier migration towards iceoryx `v2.0.0` for users.

File: doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml (hunk 4 above)
Comment (mossmaurice) [reply doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml]: @elfenpiff This list is a draft for the structure of the nextgen iceoryx. It's still under discussion if these changes will be applied for `v2.0.0`. You can run `./tools/print_all_namespace.sh` to get a list of all namespace in iceoryx.

File: README.md [hunk 9]
```diff
@@ -96,8 +96,8 @@ Please see the [Quality Declaration](./QUALITY_DECLARATION.md) for details of th
 
 |CMake project/target   | Current Level | Target Level QNX  | Target Level <br> Linux, Windows, macOS | Comment                             |
 |-----------------------|:-------------:|:-----------------:|:---------------------------------------:|:-----------------------------------:|
-| iceoryx_hoofs         | 2             | 1+                | 1                                       |                                     |
-| iceoryx_posh          | 2             | 1+, 2             | 1                                       | Will be split into separate targets |
+| iceoryx_hoofs         | 2             | 1+                | 1                                       | Except code in the namespace `aux`  |
```
Comment (mossmaurice) [inline README.md:99]: `aux` was the best idea @elBoberido and I came up with :thinking: It stands for auxiliary and shall differentiate between core vs. addon/auxiliary. Also see `tools/analyse_blacklisted_namespaces_in_binary.sh` further down.

File: iceoryx_posh/CMakeLists.txt [hunk 10]
```diff
@@ -135,16 +135,17 @@ add_library(iceoryx_posh
     source/runtime/ipc_runtime_interface.cpp
     source/runtime/ipc_message.cpp
     source/runtime/port_config_info.cpp
-    source/runtime/posh_runtime.cpp
-    source/runtime/posh_runtime_impl.cpp
-    source/runtime/posh_runtime_single_process.cpp
+    source/runtime/posh_runtime.cpp                #
+    source/runtime/posh_runtime_impl.cpp           # @todo iox-#590 These files should go into a separate library iceoryx_posh_runtime
+    source/runtime/posh_runtime_single_process.cpp #
     source/runtime/node.cpp
```
Comment (mossmaurice) [inline iceoryx_posh/CMakeLists.txt:141]: @budrus Why is the node in runtime? Does this have a specific reason?

Comment (budrus) [reply iceoryx_posh/CMakeLists.txt:141]: @mossmaurice Not really. I fear we just haven't found a better place. Feel free to suggest changes

**Review by budrus — CHANGES_REQUESTED** [review]

File: README.md [hunk 11]
```diff
@@ -64,7 +64,7 @@ You can find the full API documentation on 🌐 [https://iceoryx.io](https://ice
 | [eCAL](https://github.com/continental/ecal) | Open-source framework from [Continental AG](https://www.continental.com/) supporting pub/sub and various message protocols |
 | [RTA-VRTE](https://www.etas.com/en/products/rta-vrte.php) | [Adaptive AUTOSAR](https://www.autosar.org/standards/adaptive-platform/) platform software framework for vehicle computer from [ETAS GmbH](https://www.etas.com) |
 | [Cyclone DDS](https://github.com/eclipse-cyclonedds/cyclonedds) | Performant and robust open-source DDS implementation maintained by [ADLINK Technology Inc.](https://www.adlinktech.com/) |
-| [Apex.OS](https://www.apex.ai/apex-os) | Safe and certified software framework for autonomous mobility systems from Apex.AI |
+| [Apex.Middleware](https://www.apex.ai/apex-middleware) | Safe and certified software framework for autonomous mobility systems from [Apex.AI](https://www.apex.ai/) |
```
[the remark this answers is not in the document]
Comment (mossmaurice) [reply README.md:67]: Done.

**Review by budrus — DISMISSED** [review]

File: iceoryx_posh/CMakeLists.txt (hunk 10 above)
Comment (mossmaurice) [reply iceoryx_posh/CMakeLists.txt:141]: As just discussed, for me the node is part of a hierarchy described in [this PR](https://github.com/eclipse-iceoryx/iceoryx/pull/620/files#diff-7d0aab58cfe67a9cc98e780db81a4504ac49c4094ae5e2ba073a2b71caf91ba9R51). Let's tackle this in the follow-up PR.

File: doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+}
+
+' POSH: POSIX SHared memory
+' Publish/subscribe IPC communication infrastructure based on shared memory
+package iceoryx_posh {
+   frame "core library" {
+      component popo <<namespace>>
+      component capro <<namespace>>
+      component mepoo <<namespace>>
+      component version <<namespace>>
```
Comment (elBoberido) [inline doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml:65]: Here I also have the feeling we overdo with the namespaces. Can't we merge this with another one?

File: doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+      component runtime <<namespace>>
+   }
+
+   frame "gateway library" {
+      component gw <<namespace>>
+   }
+
+   frame "config library" {
+      component config <<namespace>>
+      component helper <<namespace>>
```
Comment (elBoberido) [inline doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml:78]: Same here. We have too many namespaces

File: doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml [hunk 14]
```diff
@@ -0,0 +1,122 @@
...
+   }
+
+   component errors <<namespace>>
+
+   frame "RouDi library" {
+      component roudi <<namespace>>
+      component build <<namespace>>
```
Comment (elBoberido) [inline doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml:85]: This is not quite correct. It is used in `iceoryx_posh_types.hpp` which is more related to the core library.

File: doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml [hunk 15]
```diff
@@ -0,0 +1,122 @@
...
+   component cpp2c <<namespace>>
+   component c2cpp <<namespace>>
+}
+
+
+' Introspection
+package iceoryx_introspection  {
+   component client <<namespace>> {
+      component introspection <<namespace>>
+   }
```
Comment (elBoberido) [inline doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml:115]: Tooooooo many namespaces :laughing:

File: doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml [hunk 16]
```diff
@@ -0,0 +1,122 @@
...
+' C-Binding
+package iceoryx_binding_c  {
+   component enums.h <<file>>
+   component event_info.h <<file>>
+   component publisher.h <<file>>
+   component subscriber.h <<file>>
+   component node.h <<file>>
+   component types.h <<file>>
+   component wait_set.h <<file>>
+   component user_trigger.h <<file>>
```
Comment (elBoberido) [inline doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml:105]: Why are the files listed? There are no files listed for the other packages

File: iceoryx_posh/CMakeLists.txt [hunk 17]
```diff
@@ -135,16 +135,17 @@ add_library(iceoryx_posh
     source/runtime/ipc_runtime_interface.cpp
     source/runtime/ipc_message.cpp
     source/runtime/port_config_info.cpp
-    source/runtime/posh_runtime.cpp
-    source/runtime/posh_runtime_impl.cpp
-    source/runtime/posh_runtime_single_process.cpp
+    source/runtime/posh_runtime.cpp                #
+    source/runtime/posh_runtime_impl.cpp           # @todo iox-#590 These files should go into a separate library iceoryx_posh_runtime
+    source/runtime/posh_runtime_single_process.cpp #
```
Comment (elBoberido) [inline iceoryx_posh/CMakeLists.txt:140]: I'm not quite sure if it makes sense to have a separate `iceoryx_posh_runtime` library. It most probably won't have many files and it is still needed by the `RouDiEnvironment`. I guess we will only know once we tried it.

**Review by elBoberido — COMMENTED** [review]
It's not quite clear to me if this shall show the current state, the desired state for v2.0 or even the state beyond v2.0.

I would assume it to be for v2.0 and therefore some parts would have to be commented out and maybe linked to a post v2.0 issue.

We overdo with some sub-namespaces and we should evaluate if we can get rid of some of them.

File: doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml [hunk 18]
```diff
@@ -0,0 +1,122 @@
...
+   backgroundColor chocolate
+}
+
+' PlantUML package: CMake package
+' PlantUML frame: CMake library
+
+' @todo #590
+' RouDi split, separate files and namespaces for static/dynamic discovery
+' Implement exported libraries (add_library in iceoryx_posh/CMakeLists.txt)
+' RouDiEnvironment shall not be in roudi namespace, but in testing
```
[the remark this answers is not in the document]
Comment (mossmaurice) [reply doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml:26]: It will be done as part of #539

File: doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml [hunk 19]
```diff
@@ -0,0 +1,122 @@
...
+header Eclipse iceoryx component overview
+
+' HOOFS: Helpful Objects Optimised For Safety
+' Library containing modern C++ STL constructs
+package iceoryx_hoofs  {
+   frame "hoofs library" {
+      component concurrent <<namespace>>
+      component cxx <<namespace>> {
+         component newtype <<namespace>>
+         component set <<namespace>>
```
[the remark this answers is not in the document]
Comment (mossmaurice) [reply doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml:37]: I have the same feeling that we overdo with the nesting of namespaces. `iox::cxx::set` is used in `iceoryx_hoofs/internal/cxx/set.hpp`

File: doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml [hunk 20]
```diff
@@ -0,0 +1,122 @@
...
+         component algorithm <<namespace>>
+      }
+      component DesignPattern <<namespace>>
```
[the remark this answers is not in the document]
Comment (mossmaurice) [reply doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml:40]: :O We can edit that in the follow-up PR that deals with the removal of the creation pattern. Once we do the `v2.0` release the `.puml` shall represent the current state.

File: doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml [hunk 21]
```diff
@@ -0,0 +1,122 @@
...
+      component errors <<namespace>>
+      component log <<namespace>> {
+         component ffbb <<namespace>>
+      }
+      component units <<namespace>> {
+         component duration_literals <<namespace>>
+
+      }
+      component posix <<namespace>>
+      component threading <<namespace>>
```
[the remark this answers is not in the document]
Comment (mossmaurice) [reply doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml:50]: Removed

File: doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml [hunk 22]
```diff
@@ -0,0 +1,122 @@
...
+
+      ' Building blocks which only meet quality level 5
+      ' Can be analysed with tools/analyse_blacklisted_namespaces_in_binary.sh
+      component aux <<namespace>>
```
[the remark this answers is not in the document]
Comment (mossmaurice) [reply doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml:54]: I would leave it here and make sure that once `v2.0` is released the `.puml` is up to date. I would also like to add it on [iceoryx.io](https://iceoryx.io) under an architecture concept article.

File: doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml (hunk 13 above)
Comment (mossmaurice) [reply doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml:78]: Agree. Would move this one level up to `iox::` same as `RouDiConfig_t`

File: doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml (hunk 12 above)
Comment (mossmaurice) [reply doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml:65]: Here, I'm would leave the namespace for `v2.0`. But that's just a gut feeling. The compatibility stuff is probably not used by a lot of users.

File: doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml (hunk 16 above)
Comment (mossmaurice) [reply doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml:105]: As C does not have namespaces I instead listed the files to give an overview over the structure.

File: iceoryx_posh/CMakeLists.txt (hunk 17 above)
Comment (mossmaurice) [reply iceoryx_posh/CMakeLists.txt:140]: I'll try thtat as a follow-up to this PR.

File: doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml (hunk 14 above)
Comment (mossmaurice) [reply doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml:85]: Moved it.

**Comment by mossmaurice** [comment]
@elBoberido 

> It's not quite clear to me if this shall show the current state, the desired state for v2.0 or even the state beyond v2.0.

I would aim to implement the `.puml` diagram for `v2.0`, update any changes to the plan shortly before the release and add the diagram to our documentation. @budrus What do you think?

File: doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml (hunk 19 above)
[the remark this answers is not in the document]
Comment (elBoberido) [reply doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml:37]: I think that this is a good rule of thumb. Too many namespaces will frustrate developers "is it foo::A or bar::A?????"

File: doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml (hunk 22 above)
[the remark this answers is not in the document]
Comment (elBoberido) [reply doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml:53]: First, please do not resolve a conversation when you disagree to give the commenter a chance to response.

Second, I did not ask for removal but to comment this out. Like you said in a previous comment, this shall show the current state and in the current state we do not have this namespace anywhere.

File: doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml (hunk 16 above)
Comment (elBoberido) [reply doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml:103]: Please remove it. It does not make sense to have the files listed. Contrary to the other components it has no structure, everything is flat and the component is binding itself.

File: doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml (hunk 22 above)
[the remark this answers is not in the document]
Comment (elBoberido) [reply doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml:53]: Oh, I would also be fine with labeling it differently. But is should not look like we currently have it.

[the remark this answers is not in the document]
Comment (mossmaurice) [reply doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml:53]: Commented it out.

File: doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml (hunk 16 above)
Comment (mossmaurice) [reply doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml:103]: You're right. There is little knowledge gain, if the files are listed. Removed them.

**Review by elBoberido — DISMISSED** [review]
The SVG looks a bit wild and I guess plantuml is not the best tool for this but we currently do not hove something better

**Review by elBoberido — APPROVED** [review]

**Review by budrus — APPROVED** [review]
